docs/_components/form-fields.md

```diff
@@ -1,22 +1,571 @@
+</fieldset>
+{% endcapture %}
+{% include helpers/code-example.html code=example %}
+
+# Form field groups
+
+{% capture example %}
+<fieldset class="usa-fieldset">
+  <legend class="usa-legend">Full name</legend>
+
+  <label for="c985" class="usa-label">First name</label>
+  <input id="c985" type="text" class="usa-input">
+
+  <label for="a2d8" class="usa-label">Last name</label>
+  <input id="a2d8" type="text" class="usa-input">
+</fieldset>
+{% endcapture %}
+{% include helpers/code-example.html code=example %}
+
+# Form validation errors
+
+{% capture example %}
+<label for="e9a4" class="usa-label">Field label</label>
+<input id="e9a4" type="text" class="usa-input usa-input-error">
+<span class="usa-input-error-message" role="alert">Error message text.</span>
+{% endcapture %}
+{% include helpers/code-example.html code=example %}
+
+## Other input types
+
+Follow the same pattern of applying `.usa-input-error` to the effected input and append the error message in the appropriate span.
+
+<div class="usa-accordion usa-accordion-bordered">
+  <h3 class="usa-accordion-heading">
+    <button class="usa-accordion-button" aria-controls="errored-textarea">
+      Textarea (multiline) input
+    </button>
+  </h3>
+  <div id="errored-textarea" class="usa-accordion-container">
+    <div class="usa-accordion-content">
+{% capture example %}
+<label for="a37c" class="usa-label">Textarea (multiline) input</label>
+<textarea id="a37c" type="text" class="usa-textarea usa-input-error"></textarea>
+<span class="usa-input-error-message" role="alert">Error message text.</span>
+{% endcapture %}
+{% include helpers/code-example.html code=example %}
+    </div>
+    <button class="usa-accordion-close-button">Close</button>
+  </div>
+
+  <h3 class="usa-accordion-heading">
+    <button class="usa-accordion-button" aria-controls="errored-dates">
+      Dates
+    </button>
+  </h3>
+  <div id="errored-dates" class="usa-accordion-container">
+    <div class="usa-accordion-content">
+{% capture example %}
+<fieldset class="usa-fieldset">
+  <legend class="usa-legend">Date of birth</legend>
+  <div class="usa-memorable-date">
+    <div class="usa-form-group usa-form-group-month">
+      <label for="e30e" class="usa-label">Month</label>
+      <input id="e30e" class="usa-input usa-input-inline usa-input-error" aria-describedby="f9e5" type="number" min="1" max="12">
+    </div>
+    <div class="usa-form-group usa-form-group-day">
+      <label for="b857" class="usa-label">Day</label>
+      <input id="b857" class="usa-input usa-input-inline usa-input-error" aria-describedby="f9e5" type="number" min="1" max="31">
+    </div>
+    <div class="usa-form-group usa-form-group-year">
+      <label for="a5dc" class="usa-label">Year</label>
+      <input id="a5dc" class="usa-input usa-input-inline usa-input-error" aria-describedby="f9e5" type="number" min="1900" max="2000">
+    </div>
+  </div>
+  <span class="usa-input-error-message" role="alert">Error message text.</span>
```

Comment:
I was thinking maybe we should include the error message icon that form fields use inside the field at the start of these error messages when they're paired with checkboxes/radios so text color isn't the only signal that something is wrong.

Reply:
Here’s what that would look like across the styles — the differences in padding and border create some interesting alignment problems. 

![Small sized radios with error message](https://user-images.githubusercontent.com/14930/51943066-958ce580-23e6-11e9-91ab-2b38676dc0e6.png)

![Normal sized radios with error message](https://user-images.githubusercontent.com/14930/51943083-a0e01100-23e6-11e9-9262-018cf92b483c.png)

![Large sized radios with error message](https://user-images.githubusercontent.com/14930/51943109-af2e2d00-23e6-11e9-969d-2b29bb26a297.png)

Thoughts? It’s definitely more usable since it’s no longer just relying on color to indicate state, but the alignment gives me pause (and seems like it’d rarely be used correctly if we added three different classes for each of these situations). 

I see why the US Web Design System went with [a simple left border](https://v2.designsystem.digital.gov/components/form-controls/#text-input) to indicate this… seems versatile (as most forms align along a simple vertical axis).

Reply:
I feel like in the long run we may want to re-think this and come up with something more sustainable (like the left border) but in the mean time I made a small adjustment of `padding-left: 1.5rem` on `.usa-input-error-message.usa-input-error-message-with-icon {` and I think it looks a bit better since the icon and text isn't really lining up with the small inputs and it just looks like a sentence with an icon in front. Let's go with this for now to keep things simple.

![screen shot 2019-01-30 at 10 54 15 am](https://user-images.githubusercontent.com/776987/51998170-c704c000-247d-11e9-8def-397c6ff1d2d1.png)
![screen shot 2019-01-30 at 10 54 01 am](https://user-images.githubusercontent.com/776987/51998171-c704c000-247d-11e9-9eb9-7d0eb46652e3.png)
![screen shot 2019-01-30 at 10 53 47 am](https://user-images.githubusercontent.com/776987/51998172-c79d5680-247d-11e9-8219-99ab303cb3a5.png)